src/main/plugins/org.dita.pdf2/xsl/fo/commons.xsl

```diff
@@ -459,6 +459,9 @@ See the accompanying license.txt file for applicable licenses.
 
     <xsl:template name="processTopicNotices">
         <fo:page-sequence master-reference="body-sequence" xsl:use-attribute-sets="page-sequence.notice">
+            <xsl:if test="key('map-id', ancestor-or-self::*[contains(@class, ' topic/topic ')][1]/@id)/ancestor::*[contains(@class,' bookmap/backmatter ')]">
```

Comment:
> The attribute-set reflection is just awful

OK, good not just me...

> Personally I think the use of attribute sets should be removed entirely...

Can't argue with that, but talk about destroying peoples' customizations...

Considering both of those comments, I think my preference would still be the first bullet above. It's a slight bit more verbose, but not out of line with the rest of the code. It remains consistent with everything else (using attribute sets to set the format), fixes the problem, and cannot interfere with existing customizations to front matter. I also don't have a strong preference for 2.3.3 versus 2.4 -- my choice would be 2.3.3 to get the fix out sooner, but if a new attribute set is seen as a new feature, I'm fine with 2.4.